CleverCloud/clever-components
# Storybook/upgrade with vite

Fixes #887 

## What does this PR do?

- removes `@web/dev-server` & `@web/dev-server-storybook` (`storybook-prebuilt`),
- installs storybook 7.0 with the `storybook/builder-vite`,
- adapts the config to match our needs.

## How to review?

- Review each commit, and focus specifically on:
  - `chore(storybook): add markdown stories`,
  - `chore(storybook): implement smart auth for smart.md stories`.
- You may skip batch modifications, I'll deal with that,
- Run storybook locally and check:
  - random stories:
    - press "d" within a story and test that the `action` and the `controls` feature still work,
    - press "i" and check that the language switching feature still works,
    - use the controls above the story and check that they still work (especially the `size` control).
  - random docs (autogenerated & md docs),
  - `cc-orga-member-list - smart` story => should contain the component working just fine at the bottom of the docs,
  - demo-smart,
  - sandbox.
- Check random stories / docs on the [preview](https://clever-components-preview.cellar-c2.services.clever-cloud.com/storybook/upgrade-with-vite/index.html?path=/docs/readme--docs) as well.

I've left a few comments (on GitHub) with notes / questions :wink: 

## TODO Flo:

- [x] remove `cc-orga-member-list.smart.md/js` modifications once the review phase is over.
- [x] dispatch feedback edits inside commits.
